SDWebImage/SDWebImageCompat.h

```diff
@@ -101,3 +102,34 @@
 #ifndef SD_UNLOCK
 #define SD_UNLOCK(lock) dispatch_semaphore_signal(lock);
 #endif
+
+#ifndef weakify
```

Comment:
Don't add these....Put all the internal macro into the private headers. Then using the pch files instead.

If we put macro here, this will effect the user application who dependency SDWebImage.

Reply:
> Don't add these....Put all the internal macro into the private headers. Then using the pch files instead.
> 
> If we put macro here, this will effect the user application who dependency SDWebImage.

Ok, I'll refactor these macros .

Reply:
@Insofan Pull the remote branch of SDWebImnage `5.x` into your fork repo's 5.x branch first.

You can use some Git GUI tools or command line for this.

Reply:
Or maybe, I can create a MR for this changes about the internal macros ? Then you can help to replace each implementation files usage.

Reply:
Actruly I have done this operation before pr. Yeah u mr a preheader file maybe a better way I don't familiar with the project structure. I will finish the rest code. The implementation as while as  I reading the source code, so it maybe a little long time. But I will do my best